client/my-sites/themes/theme-preview.jsx

```diff
@@ -103,25 +101,27 @@ const ConnectedThemePreview = connectOptions( ThemePreview );
 
 export default connect(
 	( state ) => {
-		const showPreview = isThemePreviewVisible( state );
-
-		// data for preview not set.
-		if ( ! showPreview ) {
-			return { showPreview };
+		const themeId = themePreviewVisibility( state );
+		if ( ! themeId ) {
+			return { themeId };
 		}
-		const themePreviewInfo = getThemePreviewInfo( state );
-		const themeOptions = getThemePreviewThemeOptions( state );
-		const theme = getTheme( state, themePreviewInfo.source, themePreviewInfo.themeId );
 		const siteId = getSelectedSiteId( state );
 		const isJetpack = isJetpackSite( state, siteId );
+		let theme = getTheme( state, 'wpcom', themeId );
+
+		if ( ! theme && isJetpack ) {
+			//for Jetpack sites if theme is not from wpcom we fetch demo data
+			//from wporg because data from Jetpack enpoint does not have demo_uri
+			theme = getTheme( state, 'wporg', themeId );
```

Comment:
I don't think we need this technically right now since we're [merging data from `wporg` anyway](https://github.com/Automattic/wp-calypso/blob/ef236885d8c2d17f6fe7e8269633dc0ce7851ab8/client/state/themes/selectors.js#L62-L65), so we could use `siteId` instead here.